code/obj/item/clothing/suits.dm

```diff
@@ -1283,32 +1283,55 @@
 	item_state = "space-neon"
 
 /obj/item/clothing/suit/space/custom // Used for nanofabs
+	icon_state = "spacemat"
+	inhand_image_icon = "s_suit"
+	item_state = "spacemat"
 	name = "bespoke space suit"
-	desc = "A suit that protects against low pressure environments, custom-made just for you!"
-	onMaterialChanged()
-		if(src.material)
-			if(material.hasProperty("thermal"))
-				var/prot = 100 - material.getProperty("thermal")
+	desc = "A custom built suit that protects your soft tissues from hard vacuum."
+	proc/setupSuitProp(var/datum/material/F, var/datum/material/R) // f is fabric, decides protection, R is reinforcment, decides melee prot
```

Comment:
Sorry this review will be a bit of a mess order-wise since there's a lot to cover I think (sorry).

I know we don't have a good way of handling multiple materials on an object (it's one of the projects I'd like to eventually work on but too many projects and too little time) but I think it might be wise to pick one of these materials as the base material (probably the one which you think should bestow special material properties like telecrystal teleportation or radioactivity etc.) which will be set using `.setMaterial` and keep this proc to just setting the other material.

This proc should also store the other material in some var on the space suit.

Reply:
Also the naming of the variables is bad. Why put the meaning of the variables in a comment and not in their name?

Reply:
Oh right and basically if you split this as I suggest you'd move the code which is based on the `src.masterial` back into `onMaterialChanged` as it was in the original code and keep the code relating to the second material in this proc.